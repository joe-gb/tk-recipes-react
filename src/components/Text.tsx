import React from "react";
import styled from "styled-components";

export const Text = styled.span`
  font-size: 14px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #4c5566;
`;

export const BigText = styled.span`
  font-size: 20px;
  font-weight: 500;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
`;

export const Title = styled.h1`
  font-size: 36px;
  font-weight: 600;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 44px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #1d2c3c;
  margin: 20px auto;
  max-width: 850px;
`;

export const Subtitle = styled.h2`
  font-size: 32px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 40px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #1d2c3c;
  margin: 20px auto;
  max-width: 850px;
`;

export const SmallSubtitle = styled.h3`
  font-size: 26px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 40px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #1d2c3c;
  margin: 20px auto;
  max-width: 850px;
`;
