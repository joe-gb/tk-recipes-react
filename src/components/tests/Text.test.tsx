import React from "react";
import { render } from "@testing-library/react";
import { BigText, Subtitle, Text, Title } from "../Text";

test("Text snapshot test", () => {
  let { container } = render(<Text>This is text</Text>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is text");
});

test("Bigtext snapshot test", () => {
  let { container } = render(<BigText>This is big text</BigText>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is big text");
});

test("Title snapshot test", () => {
  let { container } = render(<Title>This is a title</Title>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is a title");
});

test("Subtitle snapshot test", () => {
  let { container } = render(<Subtitle>This is a subtitle</Subtitle>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("This is a subtitle");
});
