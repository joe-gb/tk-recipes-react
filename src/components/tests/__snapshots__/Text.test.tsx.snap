// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bigtext snapshot test 1`] = `
.c0 {
  font-size: 20px;
  font-weight: 500;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
}

<div>
  <span
    class="c0"
  >
    This is big text
  </span>
</div>
`;

exports[`Subtitle snapshot test 1`] = `
.c0 {
  font-size: 32px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 40px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #1d2c3c;
  margin: 20px auto;
  max-width: 850px;
}

<div>
  <h2
    class="c0"
  >
    This is a subtitle
  </h2>
</div>
`;

exports[`Text snapshot test 1`] = `
.c0 {
  font-size: 14px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #4c5566;
}

<div>
  <span
    class="c0"
  >
    This is text
  </span>
</div>
`;

exports[`Title snapshot test 1`] = `
.c0 {
  font-size: 36px;
  font-weight: 600;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 44px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #1d2c3c;
  margin: 20px auto;
  max-width: 850px;
}

<div>
  <h1
    class="c0"
  >
    This is a title
  </h1>
</div>
`;
